### ISSUE TITLE
Incompatible to CU118 when installing causal-conv1d and mamba

### ISSUE THREAD

**Issue by wangf3014** [hustvl/Vim#21]
Thanks a lot to the authors for the wonderful work. I have some quick questions about the repo.

When installing causal-conv1d and mamba, I encountered an error saying "The detected CUDA version (12.3) mismatches the version that was used to compile PyTorch (11.8). Please make sure to use the same CUDA versions.". Do you know how to address it?

I also tried several different torch and cuda versions (eg tc211+cu121), it seems that sometimes it can be successfully installed but the model training become quite slow. Did you find similar issues about that, or does this repo have a strict constraint of the torch and cuda versions?

**Comment by pamia** [comment]
I met the same problem with you (cuda version is 12.2 + cu11.8),do you successfully train the model with  CUDA version (12.3)+ cu121

**Comment by pamia** [comment]
> Thanks a lot to the authors for the wonderful work. I have some quick questions about the repo.
> 
> When installing causal-conv1d and mamba, I encountered an error saying "The detected CUDA version (12.3) mismatches the version that was used to compile PyTorch (11.8). Please make sure to use the same CUDA versions.". Do you know how to address it?
> 
> I also tried several different torch and cuda versions (eg tc211+cu121), it seems that sometimes it can be successfully installed but the model training become quite slow. Did you find similar issues about that, or does this repo have a strict constraint of the torch and cuda versions?

I met the same problem with you (cuda version is 12.2 + cu11.8),do you successfully train the model with CUDA version (12.3)+ cu121

**Comment by Magicboomliu** [comment]
Me neither, seems this repo has a very strict constraint that the CUDA Version Must be 11.8.

**Comment by laisimiao** [comment]
> Me neither, seems this repo has a very strict constraint that the CUDA Version Must be 11.8.

I don't think so. If you install caual-conv1d and mamba whthin this repo's directory and don't use compiled whl to install, I worked on cuda11.7.

**Comment by lebron-2016** [comment]
> > Me neither, seems this repo has a very strict constraint that the CUDA Version Must be 11.8.
> 
> I don't think so. If you install caual-conv1d and mamba whthin this repo's directory and don't use compiled whl to install, I worked on cuda11.7.

Hi,  did you install the mamba library using the setup.py file in the corresponding folder? What torch version are you using under cuda11.7?
